### ISSUE TITLE
From trait implementations for common cases

### ISSUE THREAD

**Issue by vigna** [StephanvanSchaik/mmap-rs#43]
I find myself often confronted with having to turn a write-only mapping into a read-only mapping, and the `make_read_only` method, with its non-Error-implementing `Result`, is not every ergonomic in that I have to remap the error to be able to use the `?` operator.

Would you be interested in a PR with the relevant `From` trait implementation for user who do not care about getting back the original mapping in case of failure? This is just synctatic sugar, but I think it could be useful.

**Comment by StephanvanSchaik** [comment]
Yes, such a PR where the `From` trait simply drops the mapping if it fails for whatever reason sounds good, since we already have the implementation where you can get the original mapping back together with the error to fallback to.

**Comment by StephanvanSchaik** [comment]
This landed as PR #44 in mmap-rs 0.6.1.
